HHS/Head-Start-TTADP
# [Prod] Goal nudge, allow goal deletion in certain cases, improvements to admin bulk goal creation, remove dead RTTAPA code

## Description of change

- Add the frontend for goal nudge (including the ability to set a root cause and source for two grant/goal combinations at once)
- An admin using the bulk goal creation feature should have the ability to "skip" grants that already have a particular goal
- Admins & approvers can delete goals in certain cases (goals not on a report)
- This PR also removes dead code from a feature that was implemented but never "went live," the RTTAPA history reports.

## How to test

- If the CI/CD passes, the dead RTTAPA code was safely removed
- Create a goal from the RTR. You should see a "nudge" for a matching goal. If the goal is for a recipient with more than one active grant, you should have the ability to set the source variably (meaning different value per grant). If the goal is an FEI goal and the recipient has more than one active grant, you should be able to set the root causes variably.
- Create an FEI goal from the admin. Most likely, you will receive an error that tells you that the goal already exists for a given grant within the set. Use the new option to skip those grants and proceed to create for the rest of the group.
- Test that as an approver, you can delete a goal that isn't on an approved report and is "draft" or "not started"

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1942
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2612
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2567
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2157
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1995
 
## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
